src/schema.ts

```diff
@@ -1,81 +1,96 @@
+            });
         });
-    }
-
-    get models() {
-        return this.schema;
+        this.subclassMapping = subclassMapping;
     }
 
     /**
      * Check that a given class/model name exists
      */
-    has(obj) {
-        try {
-            return Boolean(this.get(obj));
-        } catch (err) {
-            return false;
-        }
+    has(obj: unknown): boolean {
+        return Boolean(this.get(obj, false));
     }
 
     /**
-     * Returns Knowledgebase class schema.
+     * Returns Knowledge base class schema.
      * @param {Object|string} obj - Record to fetch schema of.
      */
-    get(obj) {
-        let cls = obj;
+    get(obj: unknown, strict?: true): ClassDefinition;
+    get(obj: unknown, strict?: false): ClassDefinition | null;
+    get(obj: unknown, strict = true): ClassDefinition | null {
+        let className = obj;
 
         if (obj && typeof obj === 'object' && obj['@class']) {
-            cls = obj['@class'];
+            className = obj['@class'];
         }
-        return this.normalizedModelNames[typeof cls === 'string'
-            ? cls.toLowerCase()
-            : cls
-        ] || null;
+        let model: ClassDefinition | null = null;
+
+        if (typeof className === 'string') {
+            className = className.toLowerCase();
+            model = this.normalizedModelNames[className as string] || null;
+        }
+        if (!model && strict) {
+            throw new ValidationError(`Unable to retrieve model: ${className || obj}`);
+        }
+        return model;
     }
 
-    getFromRoute(routeName) {
-        for (const model of Object.values(this.schema)) {  // eslint-disable-line
+    getFromRoute(routeName: string): ClassDefinition {
+        for (const model of Object.values(this.models)) {  // eslint-disable-line
             if (model.routeName === routeName) {
                 return model;
             }
         }
         throw new Error(`Missing model corresponding to route (${routeName})`);
     }
 
-    getModels() {
-        return Object.values(this.schema);
+    getModels(): ClassDefinition[] {
+        return Object.values(this.models);
     }
 
-    getEdgeModels() {
-        return this.getModels().filter((model) => model.isEdge);
+    getEdgeModels(): ClassDefinition[] {
+        return Object.values(this.models).filter((model) => model.isEdge);
     }
 
     /**
     * Returns preview of given record based on its '@class' value
     * @param {Object} obj - Record to be parsed.
     */
-    getPreview(obj) {
-        if (obj) {
+    getPreview(obj: GraphRecord): string {
+        if (obj && typeof obj === 'object') {
             if (obj['@class'] === 'Statement') {
-                const { content } = sentenceTemplates.generateStatementSentence(this, obj);
+                const { content } = sentenceTemplates.generateStatementSentence(
+                    (arg0: Record<string, unknown>): string => this.getPreview(arg0),
```

Comment:
if this is just to prevent `getPreview` from being unbound, I would just call bind directly
```js
const { content } = sentenceTemplates.generateStatementSentence(
    this.getPreview.bind(this),
    obj,
);
```

Reply:
changed: 186add7. is there any advantage to using bind other than just making the line a little shorter?

Reply:
it's just clearer what you're trying to do. with the arrow function, it took me a minute to figure out why you even had it